oak_restricted_kernel/src/lib.rs

```diff
@@ -172,9 +172,17 @@ pub fn start_kernel(info: &BootParams) -> ! {
 
     // Note: `info` will not be valid after calling this!
     {
-        let prev_page_table = PAGE_TABLES
-            .lock()
-            .replace(mm::init_paging(program_headers).unwrap());
+        let (pml4_frame, encrypted) = mm::initial_pml4(program_headers).unwrap();
+        // Safety: the new page tables keep the identity mapping at -2GB intact, so it's safe to
+        // load the new page tables.
+        let prev_page_table = unsafe { PAGE_TABLES.lock().replace(pml4_frame, encrypted) };
+        // Prevent execution code in data only memory pages.
+        // Safety: executeable memory is assumed to be appropiately marked in the page table.
+        unsafe {
```

Comment:
I think this should happen before setting the page table in CR3.

Reply:
it does currently, cr3 is not set until L185